pkg/controller/controller.go

```diff
@@ -249,84 +249,110 @@ func (s *Service) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest
 
 	// Check if replication is enabled
 	replicationEnabled := params[s.WithRP(KeyReplicationEnabled)]
+	isMetroVolume := false
+	isMetroVolumeGroup := false
 
 	if replicationEnabled == "true" && !useNFS {
 		log.Info("Preparing volume replication")
 
-		vgPrefix, ok := params[s.WithRP(KeyReplicationVGPrefix)]
-		if !ok {
-			return nil, status.Errorf(codes.InvalidArgument, "replication enabled but no volume group prefix specified in storage class")
-		}
-		rpo, ok := params[s.WithRP(KeyReplicationRPO)]
-		if !ok {
-			return nil, status.Errorf(codes.InvalidArgument, "replication enabled but no RPO specified in storage class")
-		}
-		rpoEnum := gopowerstore.RPOEnum(rpo)
-		if err := rpoEnum.IsValid(); err != nil {
-			return nil, status.Errorf(codes.InvalidArgument, "invalid rpo value")
-		}
 		remoteSystemName, ok := params[s.WithRP(KeyReplicationRemoteSystem)]
 		if !ok {
 			return nil, status.Errorf(codes.InvalidArgument, "replication enabled but no remote system specified in storage class")
 		}
+		repMode := params[s.WithRP(KeyReplicationMode)]
+		if repMode == "" {
+			repMode = "ASYNC"
+		}
+		repMode = strings.ToUpper(repMode)
 
-		namespace := ""
-		if ignoreNS, ok := params[s.WithRP(KeyReplicationIgnoreNamespaces)]; ok && ignoreNS == "false" {
-			pvcNS, ok := params[KeyCSIPVCNamespace]
-			if ok {
-				namespace = pvcNS + "-"
+		switch repMode {
+		case "SYNC", "ASYNC":
+			// handle Sync and Async modes
+			log.Infof("%s replication mode requested", repMode)
+			vgPrefix, ok := params[s.WithRP(KeyReplicationVGPrefix)]
```

Comment:
Are we always using volume groups?

Reply:
Yes. I made a comment in the Metro section to optimize the volume group calls as we add the metro calls in later PRs.